src/notifications/use-push-notifications.tsx

```diff
@@ -110,13 +113,15 @@ export const NotificationContextProvider: React.FC = ({children}) => {
   const requestPermissions = useCallback(async () => {
     setStatus('updating');
     const permissionStatus = await requestUserPermission();
-    const token = await register(permissionStatus === 'granted');
+    // Retry to get the token after the user has granted permission
+    const token = await getFcmToken();
```

Comment:
Note that this triggers register, since `fcmToken` is in dependency list of the register function.

Reply:
So, I am putting back some of the stuff to do this retry, however the register callback it is not called himself if token changes, but it might fire other useEffect to be triggered.